transports/websocket/src/framed.rs

```diff
@@ -475,11 +477,18 @@ impl IncomingData {
         if let IncomingData::Pong(_) = self { true } else { false }
     }
 
+    pub fn is_close(&self) -> bool {
+        if let IncomingData::Closed(_) = self { true } else { false }
+    }
+
     pub fn into_bytes(self) -> Vec<u8> {
         match self {
             IncomingData::Binary(d) => d,
             IncomingData::Text(d) => d,
-            IncomingData::Pong(d) => d
+            IncomingData::Pong(d) => d,
+            IncomingData::Closed(CloseReason { code, ..}) => {
```

Comment:
What would be the use-case where one would want to transform a `Closed` into a vector of bytes? If there is none, I would prefer changing the signature of `into_bytes` to an `Option<Vec<u8>>` where `Closed` returns `None`.

Reply:
This is a wart for sure. And I guess the same argument could be made for `Pong`: what does transforming a `PONG` to bytes even mean?
I'm not sure if changing the signature helps significantly though, `into_*` methods should transform `owned –> owned` and be cheap/free. As a user I'd be a little surprised by a `into_bytes()` that returns `None`.

All of that to sum up my sentiment: shoehorning in `CLOSE` here is ugly but it already was ugly, given `PONG`. I think the real fix is to refactor things such that `into_bytes` needn't be called at all; which at the time felt a bit out of scope.

Reply:
Ouch, I went to update my substrate branch with this and got to (my own) [comment explaining why this crap is needed](https://github.com/paritytech/substrate/blob/e8c75bc526984da44f22f5d2a84191f04e4209d9/client/telemetry/src/transport.rs#L63-L67).